### ISSUE TITLE
does not work properly

### ISSUE THREAD

**Issue by AeroHand** [bmddota/ConsoleCommands#1]
i was wondering if i made anything wrong.

for example:

 FireGameEvent("console_command", {pid=0, command="dota_camera_pitch_max 25"})
       FireGameEvent("console_command", {pid=0, command="dota_camera_lock_lerp 0"})
       FireGameEvent("console_command", {pid=0, command="dota_camera_lock"})
      --FireGameEvent("console_command", {pid=0, command="dota_camera_pitch_max 10"});
       FireGameEvent("console_command", {pid=0, command="r_farz 6000"});
       FireGameEvent("console_command", {pid=0, command="unbindall"});
       FireGameEvent("console_command", {pid=0, command="alias \"+move_left\" \"LeftWalking\""});
       FireGameEvent("console_command", {pid=0, command="alias \"-move_left\" \"LeftWalkingDone\""});
       FireGameEvent("console_command", {pid=0, command="alias \"+move_right\" \"RightWalking\""});
       FireGameEvent("console_command", {pid=0, command="alias \"-move_right\" \"RightWalkingDone\""});
       FireGameEvent("console_command", {pid=0, command="bind leftarrow +move_left"});
       FireGameEvent("console_command", {pid=0, command="bind rightarrow +move_right"});

```
   FireGameEvent("console_command", {pid=0, command="alias \"+move_up\" \"UpWalking\""});
   FireGameEvent("console_command", {pid=0, command="alias \"-move_up\" \"UpWalkingDone\""});
   FireGameEvent("console_command", {pid=0, command="alias \"+move_down\" \"DownWalking\""});
   FireGameEvent("console_command", {pid=0, command="alias \"-move_down\" \"DownWalkingDone\""});
   FireGameEvent("console_command", {pid=0, command="bind uparrow +move_up"});
   FireGameEvent("console_command", {pid=0, command="bind downarrow +move_down"});

   FireGameEvent("console_command", {pid=0, command="alias \"+gasoline\" \"GasolineAcc\""});
   FireGameEvent("console_command", {pid=0, command="alias \"-gasoline\" \"GasolineAccDone\""});    
   FireGameEvent("console_command", {pid=0, command="bind space +gasoline"});
```

but the hotkey does not change and the camera does not work as I wish

thank u!

**Comment by bmddota** [comment]
Aerohand, there are 3 things that need to be kept in mind when using the ConsoleCommands.swf.

1) If the player you're sending the command to has their console open when the FireGameEvent is received, it will not successfully execute. In fact, it will actually ensure that all future "console_command" messages fail to that player.  It may be possible to implement a queue/retry system into the .swf so that as soon as they close their console I resend all of the commands, but for now the console has to be closed for it to work.

2) They have to be in the game and loaded, so sending a command anytime before the player_connect_full (or possibly the start of the HERO_SELECTION game state) may not work.

3) If you do successfully send the console_command with their console not open, it will "lock" them out of using the console entirely for the duration of the game.  This is annoying when doing testing in tools, so once you get it working in a general case, you should probably put in a conditional which checks if you're running the tools and if so uses SendToConsole instead of the "console_command" event.  This way you can keep testing and using the console in tools mode, and then have it use "console_command" when live for multiplayer.

**Comment by AeroHand** [comment]
so u mean if i wanna test my code in workshop tool i shouldn't open the console?
i tried so but it still did not work...

**Comment by bmddota** [comment]
Yeah you can't open the console and you have to fire the game events after you're in the game.
